test/Conversion/TorchToLinalg/onehot.mlir

```diff
@@ -0,0 +1,42 @@
+// RUN: torch-mlir-opt --pass-pipeline="builtin.module(func.func(torch-decompose-complex-ops),torch-backend-to-linalg-on-tensors-backend-pipeline)" -verify-diagnostics < %s | FileCheck %s
```

Comment:
I have not been involved with the ONNX work, but the test here looks like it's showing an integration test for changes in multiple places. I would expect to see a single test for the `AtenIntScalar` op lowering to arith added in this PR, and then separate tests for how the `onnx.one_hot` -> `torch.aten.one_hot` changed, and for how the `torch.aten.one_hot` decomposition changed, rather than testing all the way to linalg in one shot for this specific case.

There is a lit test file for testing changes to `DecomposeComplexOps` here: https://github.com/llvm/torch-mlir/blob/main/test/Dialect/Torch/decompose-complex-ops.mlir

And it looks to me that the `aten.scalar.int` to Arith test can go here: https://github.com/llvm/torch-mlir/blob/main/test/Conversion/TorchToArith/basic.mlir

The onnx test (if needed) can go here: https://github.com/llvm/torch-mlir/blob/main/test/Conversion/TorchOnnxToTorch/simple_ops_g_to_p.mlir